src/lib/certmap/sss_cert_content_crypto.c

```diff
@@ -707,6 +707,75 @@ static int get_extended_key_usage_oids(TALLOC_CTX *mem_ctx,
     return ret;
 }
 
+static int get_serial_number(TALLOC_CTX *mem_ctx, X509 *cert,
+                             uint8_t **serial_number,
+                             size_t *serial_number_size,
+                             const char **serial_number_dec_str)
+{
+    const ASN1_INTEGER *serial;
+    BIGNUM *bn = NULL;
+    size_t size;
+    uint8_t *buf = NULL;
+    int ret;
+    char *tmp_str = NULL;
+
+    serial = X509_get0_serialNumber(cert);
+    bn = ASN1_INTEGER_to_BN(serial, NULL);
+    if (bn == NULL) {
+        *serial_number_size = 0;
+        *serial_number = NULL;
+        return 0;
+    }
+
+    /* The serial number MUST be a positive integer. */
+    if (BN_is_zero(bn)) {
```

Comment:
Technically `BIGNUM` can also be negative.
But check using `BN_cmp()` and `BN_zero()` would be annoying...

Reply:
Huh, I was pretty sure it should exist but was confused by `No manual entry for BN_is_negative`.